src/Features/Core/Portable/ConvertIfToSwitch/AbstractConvertIfToSwitchCodeRefactoringProvider.Analyzer.cs

```diff
@@ -0,0 +1,397 @@
+            }
+
+            // Tree to parse:
+            //
+            //    <section-expr>
+            //        : <section-expr> || <pattern-expr>
+            //        | <pattern-expr>
+            //
+            private bool ParseSwitchLabels(IOperation operation, ArrayBuilder<AnalyzedSwitchLabel> labels)
+            {
+                if (operation is IBinaryOperation { OperatorKind: ConditionalOr } op)
+                {
+                    if (!ParseSwitchLabels(op.LeftOperand, labels))
+                    {
+                        return false;
+                    }
+
+                    operation = op.RightOperand;
+                }
+
+                var label = ParseSwitchLabel(operation);
+                if (label is null)
+                {
+                    return false;
+                }
+
+                labels.Add(label);
+                return true;
+            }
+
+            private AnalyzedSwitchLabel? ParseSwitchLabel(IOperation operation)
+            {
+                var guards = ArrayBuilder<TExpressionSyntax>.GetInstance();
+                var pattern = ParsePattern(operation, guards);
+                if (pattern is null)
+                {
+                    guards.Free();
+                    return null;
+                }
+
+                return new AnalyzedSwitchLabel(pattern, guards.ToImmutableAndFree());
+            }
+
+            private enum ConstantResult
+            {
+                None,
+                Left,
+                Right,
+            }
+
+            private ConstantResult DetermineConstant(IBinaryOperation op)
+            {
+                return (op.LeftOperand, op.RightOperand) switch
+                {
+                    var (e, v) when IsConstant(v) && CheckTargetExpression(e) => ConstantResult.Right,
+                    var (v, e) when IsConstant(v) && CheckTargetExpression(e) => ConstantResult.Left,
+                    _ => ConstantResult.None,
+                };
+            }
+
+            // Tree to parse:
+            //
+            //    <pattern-expr>
+            //        : <pattern-expr> && <expr>                         // C#
+            //        | <expr0> is <pattern>                             // C#
+            //        | <expr0> is <type>                                // C#
+            //        | <expr0> == <const-expr>                          // C#, VB
+            //        | <expr0> <comparison-op> <const>                  //     VB
+            //        | ( <expr0> >= <const> | <const> <= <expr0> )
+            //           && ( <expr0> <= <const> | <const> >= <expr0> )  //     VB
+            //        | ( <expr0> <= <const> | <const> >= <expr0> )
+            //           && ( <expr0> >= <const> | <const> <= <expr0> )  //     VB
+            //
+            private AnalyzedPattern? ParsePattern(IOperation operation, ArrayBuilder<TExpressionSyntax> guards)
+            {
+                switch (operation)
```

Comment:
possibly simpler as switch expression.

Reply:
not possible. there's a statement in there.